src/ServiceInsight/MessageFlow/MessageFlowViewModel.cs

```diff
@@ -118,7 +118,11 @@ public void ShowSagaWindow()
             {
                 selection.SelectedMessage = SelectedMessage.Message;
             }
-            eventAggregator.PublishOnUIThread(SwitchToSagaWindow.Instance);
+
+            if (selection.SelectedMessage?.HasOneSaga == true)
```

Comment:
This is because if you have two sagas we don't know which one to show?

Reply:
@mikeminutillo Yes. In Flow Diagram, you could click on the saga part and it'd take you to the view showing the incorrect saga. This can be revisited once we have a way of showing multiple sagas in the Saga View.

Reply:
Would it be better to show the first one? The Saga View will still work for it and that keeps the functionality the same as it is today.

Reply:
@mikeminutillo 	agreed. Rolled back these bits so it is back to its original functionality, although I find it confusing.